pkg1/elastic/elasticsearch.go

```diff
@@ -0,0 +1,190 @@
+package elastic
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"logexplorationapi/pkg1/constants"
+	"logexplorationapi/pkg1/logs"
+	"net/http"
+	"strings"
+)
+
+type ElasticRepository struct {
+	esClient *elasticsearch.Client
+}
+
+func InitializeElasticSearchClient() *elasticsearch.Client {
+
+	esAddress := flag.String("es-address", "https://localhost:9200", "ElasticSearch Server Address")
+	esCert := flag.String("es-cert", "admin-cert", "admin-cert file location")
+	esKey := flag.String("es-key", "admin-key", "admin-key file location")
+	flag.Parse() //fetch command line parameters
+	cert, err := tls.LoadX509KeyPair(*esCert, *esKey)
+	if err != nil {
+		fmt.Println(err)
+	}
+	cfg := elasticsearch.Config{
+		Addresses: []string{
+			*esAddress,
+		},
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true,
+				Certificates: []tls.Certificate{cert}},
+		},
+	}
+	esClient, err := elasticsearch.NewClient(cfg)
+	if err != nil {
+		fmt.Println("Error", err)
+	}
+
+	return esClient
+}
+
+func NewElasticRepository() logs.LogsProvider {
```

Comment:
I think you can move your ES init logic into this method and pass all the required parameters (`es-host, `es-cert`, etc.) as arguments.